godtossing/Pos/Ssc/GodTossing/Core/Types.hs

```diff
@@ -244,11 +248,31 @@ getCertId = addressHash . vcSigningKey
 
 -- | VssCertificatesMap contains all valid certificates collected
 -- during some period of time.
-type VssCertificatesMap = HashMap StakeholderId VssCertificate
+newtype VssCertificatesMap = VssCertificatesMap
+    { getVssCertificatesMap :: HashMap StakeholderId VssCertificate }
+    deriving (Eq, Show, Generic, NFData, Monoid, Container, NontrivialContainer)
 
--- | Safe constructor of 'VssCertificatesMap'. TODO: wrap into newtype.
+type instance Element VssCertificatesMap = VssCertificate
+
+memberVss :: StakeholderId -> VssCertificatesMap -> Bool
+memberVss id (VssCertificatesMap m) = HM.member id m
+
+lookupVss :: StakeholderId -> VssCertificatesMap -> Maybe VssCertificate
+lookupVss id (VssCertificatesMap m) = HM.lookup id m
+
+insertVss :: VssCertificate -> VssCertificatesMap -> VssCertificatesMap
+insertVss c (VssCertificatesMap m) =
+    VssCertificatesMap (HM.insert (getCertId c) c m)
+
+deleteVss :: StakeholderId -> VssCertificatesMap -> VssCertificatesMap
+deleteVss id (VssCertificatesMap m) = VssCertificatesMap (HM.delete id m)
+
+-- | Safe constructor of 'VssCertificatesMap'.
+--
+-- (Well, almost safe: it doesn't check that all certificates have different
+-- 'vcVssKey's).
```

Comment:
😔

Reply:
Perhaps making it safe **and** total is too much for now, but I would make it partial then and use `error` if invariant is violated. I believe it's safer. And this check should be cheap, because we won't have many participants in GodTossing anyway.